client/src/main/java/io/pravega/client/stream/impl/PositionImpl.java

```diff
@@ -112,6 +131,35 @@ private void write00(PositionImpl position, RevisionDataOutput revisionDataOutpu
             revisionDataOutput.writeMap(map, (out, s) -> out.writeUTF(s.getScopedName()),
                                         (out, offset) -> out.writeCompactLong(offset));
         }
+        
+        private void read01(RevisionDataInput revisionDataInput, PositionBuilder builder) throws IOException {
+            Map<Segment, Range> map = revisionDataInput.readMap(in -> Segment.fromScopedName(in.readUTF()), PositionSerializer::readRange);
+            builder.segmentRanges(map);
```

Comment:
this does not update `ownedSegments ` whereas `public PositionImpl(Map<SegmentWithRange, Long> segments) ` does update `ownedSegments`

Reply:
This is a deserialization method. As it is a revision and not a version the deserialization process is to call read00() and then read01(). ownedSegments is updated in read00()